docs/_posts/2020-02-28-puppetwash.md

```diff
@@ -0,0 +1,283 @@
+        Subject: CN=puppetdb.test
+```
+So we should be able to connect to PuppetDB using it's default port 8081 at `puppetdb.test`. We can confirm that with `cat puppetdb.conf`. To get the local port we'll use to connect to it - within the Pupperware project - run
+```
+$ docker-compose port puppetdb 8081
+0.0.0.0:32770
+```
+
+Let's copy the certificates somewhere we can use them later. The Puppet Server will be in PuppetDB's whitelist, so let's use that one since it's easily accessible.
+```
+wash docker/volumes/pupperware_puppetserver-config ❯ mkdir -p ~/.puppetlabs/wash/pupperware
+wash docker/volumes/pupperware_puppetserver-config ❯ cp certs/ca.pem certs/puppet.test.pem ~/.puppetlabs/wash/pupperware/
+wash docker/volumes/pupperware_puppetserver-config ❯ cp private_keys/puppet.test.pem ~/.puppetlabs/wash/pupperware/puppet.test.key.pem
+```
+
+Last, let's make sure there's something in PuppetDB to look at. In the Pupperware project we'll do a noop agent run to add facts and a report
+```
+$ docker-compose exec puppet puppet agent -t --noop
+Info: Using configured environment 'production'
+Info: Retrieving pluginfacts
+Info: Retrieving plugin
+Info: Retrieving locales
+Info: Applying configuration version '1582848507'
+Info: Creating state file /opt/puppetlabs/puppet/cache/state/state.yaml
+Notice: Applied catalog in 0.02 seconds
+```
+
+## Configure Puppetwash
+
+To configure Puppetwash we need to install and configure it. It's distributed as a Ruby Gem, so we can install it with
+```
+gem install puppetwash
+```
+
+Get the path to the Puppetwash script with
+```
+$ gem contents puppetwash
+/Users/me/.gem/ruby/2.6.0/gems/puppetwash-0.2.0/puppet.rb
+```
+
+We then add that to Wash's config at `~/.puppetlabs/wash/wash.yaml`
+```
+external-plugins:
+- script: '/Users/me/.gem/ruby/2.6.0/gems/puppetwash-0.2.0/puppet.rb'
+```
+
+If we were using Puppet Enterprise we would use a user authentication (RBAC) token. Wash config would look like
+```
+external-plugins:
+- script: '/Users/me/.gem/ruby/2.6.0/gems/puppetwash-0.2.0/puppet.rb'
+my_pe_instance:
+  puppetdb_url: https://puppet:8081
+  rbac_token: <my_rbac_token>
+  cacert: /path/to/cacert.pem
+```
+
+With open-source Puppet, we'll use cert-based authentication. Using the examples from above, `~/.puppetlabs/wash/wash.yaml` should look like
+```
+external-plugins:
+- script: '/Users/me/.gem/ruby/2.6.0/gems/puppetwash-0.2.0/puppet.rb'
+my_pe_instance:
+  puppetdb_url: https://puppetdb.test:32770
+  cacert: /Users/michaelsmith/.puppetlabs/wash/pupperware/ca.pem
+  cert: /Users/michaelsmith/.puppetlabs/wash/pupperware/puppet.test.pem
+  key: /Users/michaelsmith/.puppetlabs/wash/pupperware/puppet.test.key.pem
+```
+
+Start (or restart) Wash to load the new Puppetwash config and you should now be able to explore PuppetDB data in Wash
+```
+wash . > cd puppet/pupperware/nodes
+wash puppet/pupperware/nodes > tree
```

Comment:
I think pasting `stree`'s output before this would be useful for that high-level overview of what's in this plugin. I think a narrative similar to https://puppetlabs.github.io/wash/2020-02-07-boltwash might be nice, where you start by incrementally ls'ing through the plugin ("these are my instances", "these are my nodes"). I think the latter half of what you're showing off is OK though

Reply:
So basically ls some stuff before the `tree`